idea/src/org/jetbrains/kotlin/idea/quickfix/SuperClassNotInitialized.kt

```diff
@@ -61,21 +62,29 @@ object SuperClassNotInitialized : KotlinIntentionActionsFactory() {
 
         val superClass = (type.constructor.declarationDescriptor as? ClassDescriptor) ?: return emptyList()
         val classDescriptor = classOrObjectDeclaration.resolveToDescriptorIfAny(BodyResolveMode.FULL) ?: return emptyList()
-        val constructors = superClass.constructors.filter { it.isVisible(classDescriptor) }
+        val inSameDeclaration = superClass.containingDeclaration == classDescriptor.containingDeclaration
```

Comment:
It's not enough. Both declarations must be either top-level or class inside super-class (the last case is covered by isVisible). Otherwise you cannot inherit sealed class.

Reply:
I fix it myself